src/kafka/inflight_activation_writer.rs

```diff
@@ -73,6 +77,17 @@ impl Reducer for InflightActivationWriter {
             return Ok(None);
         }
 
+        if self
+            .store
+            .count_by_status(InflightActivationStatus::Delay)
+            .await
+            .expect("Error communicating with activation store")
+            + batch.len()
```

Comment:
I don't think we want to limit purely by `batch.len` here. This could mean that a batch that contains no delay tasks will get blocked. I think the batch should only backpressure if it is actually adding more delay tasks.

Reply:
I'm more concerned about this case then the case of delayed tasks being blocked.

If we have to pass the numbers through when the batch is created then let's do that. I don't want it to be possible for ingestion tasks to stop because of some delayed task.

Reply:
Okay I've updated this PR to scan for delay and pending tasks in the reduce step. Then, depending on what is currently sitting in sqlite, only write the activations that make the limit exceeded. This allows pending tasks to be written even when delayed task are at max capacity. I've also added some tests. I'll need a re-review on this